core/src/env/srml/types.rs

```diff
@@ -31,7 +31,7 @@ impl EnvTypes for DefaultSrmlTypes {
 
 /// The default SRML address type.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Encode, Decode)]
-pub struct Address([u8; 32]);
+pub struct Address(pub [u8; 32]);
```

Comment:
I am not sure about `pub` here (same in `Hash`). Generally for the sake of encapsulation and type safety I try to avoid using `pub` fields, especially in tuple structs. The reason for this is: If the inner field is private we cannot for example convert a `Hash` easily into an `Address` instance - thus avoiding the strict type system. If a users receives an `Address` he should only be able to use it as such. On the other hand this might be a bit restrictive in the general sense so as a backdoor for users that know what they are doing we would implement some methods to receive their internals.

Reply:
I needed to do this in order to construct the `Address` and `Hash` type in the `test_env`.

Like so:
```
self.caller = srml::Address([0; 32]);
```

Otherwise, I guess I could implement a public constructor?

Reply:
> Otherwise, I guess I could implement a public constructor?

Reply:
Thanks for taking the time to work with me here :)

Latest commit adds public constructors `new()` and uses them to initialize these types from `[u8; 32]`